### PR TITLE
Change password logout

### PR DESCRIPTION
**Pull request by TijanAyo**
New branch... since the old one had serious merge conflict with the main branch

### CODE REVIEW

File: src/auth/auth.controller.ts [hunk 1]
```diff
@@ -36,6 +38,20 @@ export class AuthController {
     return ZaLaResponse.Ok(userSignIn, 'Successfully logged in', 201);
   }
 
+  @Post('/signout')
+  @ApiOperation({description: 'Sign out a user'})
+  async signOutUser(@Body('refreshToken') refresToken:string) {
```
Comment (Chimexx) [inline src/auth/auth.controller.ts:43]: Nice work, just one problem.

There is no field to enter the refresh token, fix your line 43

**Review by Chimexx — CHANGES_REQUESTED** [review]

File: src/auth/auth.controller.ts (hunk 1 above)
Comment (TijanAyo) [reply src/auth/auth.controller.ts:43]: No field ??

I did test before I pushed... I used postman to test each endpoint before making this PR.

Comment (Chimexx) [reply src/auth/auth.controller.ts:43]: So where did you enter the token? as a body or as param? You have body but no Dto, and the field you are extracting from the body `refreshToken` doesn't match with variable name `refresToken`. Yet you insist you tested this? hmmm.

Comment (TijanAyo) [reply src/auth/auth.controller.ts]: Was working on another branch...

**Review by Chimexx — APPROVED** [review]

**Review by SunmadeBabafemi — APPROVED** [review]

**Comment by SunmadeBabafemi** [comment]
i made a few changes... on the PR sequel to Chimexx's commit... it all works fine

**Review by johnayinde — APPROVED** [review]

File: src/auth/auth.service.ts [hunk 2]
```diff
@@ -96,6 +102,29 @@ export class AuthService {
...
+
+      if (!user) {
+        throw new BadRequestException(
+          ZaLaResponse.BadRequest(
+            'Invalid Refresh Token',
+            'Get the correct refresh token and try again',
+          ),
+        );
+      }
+      await this.userRepo.update(user.id, { refreshToken: null });
```
Comment (PhilemonBrain) [inline src/auth/auth.service.ts:120]: change to expiration date to last 5mins

**Review by PhilemonBrain — CHANGES_REQUESTED** [review]

**Pull request merged**